invenio_app_ils/ui/src/invenio_app_ils/common/components/ResultsTable/formatters.js

```diff
@@ -1,41 +1,99 @@
+    'Transaction date': toShortDateTime(
+      fromISO(loan.metadata.transaction_date)
+    ),
+    'Expiration date': toShortDateTime(
+      fromISO(loan.metadata.request_expire_date)
+    ),
+    Renewals: loan.metadata.extension_count,
   };
 }
 
 function formatDocumentToTableView(document) {
-  const serialized = {
-    ID: document.document_pid,
-    Title: document.title,
+  let serialized = {
+    ID: document.document_pid ? document.document_pid : document.id,
+    Created: toShortDate(fromISO(document.created)),
+    Updated: toShortDate(fromISO(document.updated)),
   };
-  if (document.circulation) {
-    serialized['Loan requests'] = document.circulation.pending_loans;
-    serialized['Loanable items'] = document.circulation.loanable_items;
+  if (!_isEmpty(document.metadata)) {
+    serialized['Title'] = document.metadata.title;
+    serialized['Authors'] = document.metadata.authors;
+    if (!_isEmpty(document.metadata.circulation)) {
+      serialized['Requests'] = document.metadata.circulation.pending_loans;
+      serialized['Items'] = document.metadata.circulation.loanable_items;
+    }
   }
   return serialized;
 }
 
 function formatItemToTableView(item) {
   return {
-    ID: item.item_pid,
-    Barcode: item.barcode,
-    'Document ID': item.document_pid,
-    Status: item.status,
-    'Circulation status': item.circulation_status,
-    'Internal location': item.internal_location,
+    ID: item.item_pid ? item.item_pid : item.id,
     Created: toShortDate(fromISO(item.created)),
     Updated: toShortDate(fromISO(item.updated)),
+    Barcode: item.metadata.barcode,
+    'Document ID': item.metadata.document_pid,
+    Status: item.metadata.status,
+    Medium: item.metadata.medium,
+    'Circulation status': item.metadata.circulation_status
+      ? item.metadata.circulation_status.state
+      : null,
+    'Internal location': item.metadata.internal_location
+      ? item.metadata.internal_location.name
+      : null,
+    Location: item.metadata.internal_location.location
+      ? item.metadata.internal_location.location.name
+      : null,
+    Shelf: item.metadata.shelf,
   };
 }
 
+function formatLocationToTableView(location) {
+  return {
+    ID: location.location_pid ? location.location_pid : location.id,
+    Created: toShortDate(fromISO(location.created)),
+    Updated: toShortDate(fromISO(location.updated)),
+    Link: location.links.self,
+    Name: location.metadata.name,
+    Address: location.metadata.address,
+    Email: location.metadata.email,
+  };
+}
+
+function formatInternalLocationToTableView(internal_loc) {
+  let entry = {
+    ID: internal_loc.item_pid ? internal_loc.item_pid : internal_loc.id,
+    Created: toShortDate(fromISO(internal_loc.created)),
+    Updated: toShortDate(fromISO(internal_loc.updated)),
+    Link: internal_loc.links.self,
+  };
+  if (!_isEmpty(internal_loc.metadata)) {
+    entry['Name'] = internal_loc.metadata.name;
```

Comment:
We can also write it like this (I just noticed you wrote it like this above), but I leave it to you to decide.
```
entry = {
  'Name': internal_loc.metadata.name,
  'Physical location': internal_loc.metadata.physical_location,
  'Location e-mail': internal_loc.metadata.location.email,
  'Location name': internal_loc.metadata.location.name,
  'Location id': internal_loc.metadata.location.location_pid,
}
```